### PR TITLE
openid connector support

### PR DESCRIPTION
**Pull request by leena-jawale**
Signed-off-by: Leena Jawale <ljawale@redhat.com>

Added OpenID connector support 
Updated dexserver api to add openid connector
Updated unit tests

### CODE REVIEW

**Comment by leena-jawale** [comment]
Once this review and approved, I'll update go.mod in my next PR

File: hack/generate-cr.sh
Comment (robinbobbitt) [inline hack/generate-cr.sh:157]: I'm confused by what we're representing here. It looks like the dex server will be pointing at itself as an OIDC provider. Shouldn't we be asking for the user to provide the OIDC Issuer URL just as they provide a client id and secret?

File: controllers/dexserver_controller.go
Comment (robinbobbitt) [inline controllers/dexserver_controller.go:386]: Should/can we refactor this given that the code looks the same for all connector types other than secret name and which data from the secret is returned?

File: api/v1alpha1/dexserver_types.go
Comment (robinbobbitt) [inline api/v1alpha1/dexserver_types.go:191]: Nit: Let's fix "ConnectorTypeOpenID" => "ConnectorTypeOIDC" while we're in here

**Review by robinbobbitt — COMMENTED** [review]
Looks great! Just a few things to check on.

File: hack/generate-cr.sh
Comment (leena-jawale) [reply hack/generate-cr.sh:157]: you're right.. user will provide issuer URL

File: go.mod
Comment (leena-jawale) [reply go.mod:125]: Keeping it for this PR. Will update go.mod for this in the next PR. Not sure how you people handle this. Do you first commit api changes and then make code changes?

File: controllers/dexserver_controller.go
Comment (leena-jawale) [reply controllers/dexserver_controller.go:386]: Secret name is different for each connector. Even if I refactor this logic, still I'll have to add switch for secret name. I think we should keep it as it is. thoughts?

File: go.mod
Comment (robinbobbitt) [reply go.mod:125]: I'm admittedly not anywhere near expert on go modules, but I'm not sure why you'd need this in here at all. It seems like it's saying to use your copy of the code rather than what's in the local repo itself. What breaks if you just take this out?

File: controllers/dexserver_controller.go
Comment (robinbobbitt) [reply controllers/dexserver_controller.go:386]: I see the check for namespace is different in each case as well, so maybe not quite to the point that we need to refactor.

File: go.mod
Comment (leena-jawale) [reply go.mod:125]: This PR is not yet merged. So api changes are not in local copy ‘identitatem/dex-operator’. Api changes are still in my forked repo branch. I was thinking we can merge this PR and after merge will open another PR to remove this line and update go.mod..Another option is to make separate PR for api change. Once that merge I’ll make another PR for code change. Which option sounds viable to you?

File: controllers/dexserver_controller.go
Comment (vidyanambiar) [inline controllers/dexserver_controller.go:1013]: Just an FYI - I realized that https://github.com/identitatem/dex-operator/pull/95 is not merged yet and fell off my radar 😞 . Sensitive values like client secret need to be provided to the configMap as environment variables. Just a heads up that this may need to be updated for the OIDC connector when https://github.com/identitatem/dex-operator/pull/95 is merged.

Comment (leena-jawale) [reply controllers/dexserver_controller.go:1013]: Yeah I'll update for oidc once #95 is merged

**Review by robinbobbitt — APPROVED** [review]
Thanks @leena-jawale ! lgtm

**Comment by chrisahl** [comment]
Part of https://github.com/stolostron/backlog/issues/19070

**Pull request merged**
